docs/src/documentation/03-components/01-action/textlink/03-accessibility.mdx

```diff
@@ -0,0 +1,116 @@
+
+### Automatic Accessibility Features
+
+The TextLink component automatically manages several accessibility features:
+
+- The component automatically manages HTML semantics and ARIA attributes:
+
+  - Uses the semantically correct element based on props:
+    - Renders as `<a>` when `href` is provided
+    - Renders as `<button>` when no `href` is provided
+    - Renders as specified custom element when `asComponent` is provided
+  - Sets appropriate ARIA states based on component props
+  - Sets `role="button"` when `onClick` is provided without `href`
+  - Sets proper `rel` attributes for external links (`noopener`, `noreferrer`)
+  - Sets `type="button"` when rendered as a button element
+
+- Focus management is handled automatically:
+  - Native focus behavior is preserved based on the rendered element
+  - Ensures proper focus appearance for keyboard users
+
+### Best Practices
+
+- Use clear and descriptive link text that describes the destination or purpose of the link. Avoid generic text like "click here" or "read more" without proper context.
+
+- When using icons in links (with `iconLeft` or `iconRight`), ensure they have appropriate accessibility labels:
+
+  - If the icon is decorative and accompanied by text, the icon should be hidden from screen readers
+  - If the icon alone conveys meaning (especially when used without text), use `aria-label` on the icon to describe its purpose
+
+- For links that open in a new window/tab (when `external={true}`), ensure users are aware of this behavior:
+
+  - Always use the `<NewWindow />` icon to visually indicate that the link opens in a new window/tab
+  - Add an accessibility label to the icon using `ariaLabel="Opens in new window"` to ensure screen reader users are informed
+
+- For TextLinks with only icons (no text content), always provide an `aria-label` to ensure screen reader users understand the link's purpose.
+
+- The `title` attribute is not reliably announced by screen readers and should not be used as the primary method for providing additional information to screen reader users. Instead:
+  - Use descriptive link text
+  - Use `aria-label` for links with unclear text or icon-only links
+  - For additional descriptive information, consider using visible text or alternative ARIA attributes
+
+### Keyboard Navigation
+
+TextLink components are navigable with keyboard by default:
+
+- **When rendered as anchor (`<a>`):**
+
+  - Users can focus on links using the **Tab** key
+  - Users can activate links using the **Enter** key
+
+- **When rendered as button (`<button>`):**
+  - Users can focus using the **Tab** key
+  - Users can activate using both **Enter** and **Space** keys
+
+On mobile devices with on-screen keyboards, standard touch interactions apply while maintaining the same keyboard accessibility patterns when an external keyboard is connected.
+
+### Examples
```

Comment:
It would be good to have examples for ariaCurrent and aria-controls (or at least the first one)